packages/e2e/test/wallet/SingleAddressWallet/delegation-rewards.test.ts

```diff
@@ -0,0 +1,227 @@
+import * as envalid from 'envalid';
+import { Cardano, createSlotEpochCalc } from '@cardano-sdk/core';
+import { FaucetProvider, TestWallet, faucetProviderFactory, getWallet } from '../../../src';
+import { SignedTx, buildTx } from '@cardano-sdk/wallet';
+import { assertTxIsValid } from '../../../../wallet/test/util';
+import { filter, firstValueFrom, map, take } from 'rxjs';
+import { isNotNil } from '@cardano-sdk/util';
+import { logger } from '@cardano-sdk/util-dev';
+
+// Verify environment.
+export const env = envalid.cleanEnv(process.env, {
+  ASSET_PROVIDER: envalid.str(),
+  ASSET_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  CHAIN_HISTORY_PROVIDER: envalid.str(),
+  CHAIN_HISTORY_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  FAUCET_PROVIDER: envalid.str(),
+  FAUCET_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  KEY_MANAGEMENT_PARAMS: envalid.json({ default: {} }),
+  KEY_MANAGEMENT_PROVIDER: envalid.str(),
+  LOGGER_MIN_SEVERITY: envalid.str({ default: 'info' }),
+  NETWORK_INFO_PROVIDER: envalid.str(),
+  NETWORK_INFO_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  REWARDS_PROVIDER: envalid.str(),
+  REWARDS_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  STAKE_POOL_PROVIDER: envalid.str(),
+  STAKE_POOL_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  TX_SUBMIT_PROVIDER: envalid.str(),
+  TX_SUBMIT_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  UTXO_PROVIDER: envalid.str(),
+  UTXO_PROVIDER_PARAMS: envalid.json({ default: {} })
+});
+
+describe('delegation rewards', () => {
+  let faucetProvider: FaucetProvider;
+  let wallet1: TestWallet;
+  let wallet2: TestWallet;
+
+  beforeAll(async () => {
+    faucetProvider = await faucetProviderFactory.create(env.FAUCET_PROVIDER, env.FAUCET_PROVIDER_PARAMS, logger);
+
+    await faucetProvider.start();
+
+    const healthCheck = await faucetProvider.healthCheck();
+
+    if (!healthCheck.ok) throw new Error('Faucet provider could not be started.');
+  });
+
+  afterAll(async () => {
+    await faucetProvider.close();
+  });
+
+  beforeEach(async () => {
+    wallet1 = await getWallet({ env, logger, name: 'Sending Wallet', polling: { interval: 50 } });
+    wallet2 = await getWallet({ env, logger, name: 'Receiving Wallet', polling: { interval: 50 } });
```

Comment:
Other props don't seem to be used
```suggestion
    ({ wallet: wallet1 }) = await getWallet({ env, logger, name: 'Sending Wallet', polling: { interval: 50 } });
    ({ wallet: wallet2 }) = await getWallet({ env, logger, name: 'Receiving Wallet', polling: { interval: 50 } });
```

Reply:
`wallet1.providers` is used a couple of times... but good suggestion! Thank you